pkg/types/chain_reader.go

```diff
@@ -51,21 +68,212 @@ type BoundContract struct {
 	Pending bool
 }
 
-type Event struct {
-	ChainID           string
-	EventIndexInBlock string
-	Address           string
-	TxHash            string
-	BlockHash         string
-	BlockNumber       int64
-	BlockTimestamp    time.Time
-	CreatedAt         time.Time
-	Keys              []string
-	Data              []byte
+type Sequence struct {
+	// TODO SequenceCursor, this should be a unique sequence identifier that chain reader impl. understands.
+	// This way we can retrieve past/future sequences (EVM log events) very granularly but still hiding the chain detail.
+	SequenceCursor string
```

Comment:
Will the `SequenceCursor` be required to be unique? Is it made of a block number or a combination of data?

Reply:
It will be unique since its a combination of different things that make a sequence(log event) unique on that chain